syntax = "proto3";
import "google/protobuf/empty.proto";


package message;

message Message{
    uint64 id = 1;
    uint64 user_id = 2;
    uint64 chat_id = 3;
    string message = 4;
    MessageUser user = 5;

};

message CreateMessageRequest{
    uint64 chat_id = 1;
    string message = 2;
    uint64 user_id = 3;
    string token = 4;

    string username = 5;
    string profile_picture = 6;
    string name = 7;


};


message  CreateMessageResponse{
    Message message = 1;
};


message GetMessagesResponse{
    repeated Message message = 1;
};

message GetMessagesRequest{
    uint64 chat_id = 1;
    uint64 user_id = 2;
    string token = 3;
};


message MessageUser{
    
    uint64 id = 1;
    string username = 2;
    string profile_picture = 3;
    string name = 4;
    string surname = 5;
};

message DeleteMessageRequest{
    uint64 id = 1;
    uint64 user_id = 2;
};


message UpdateMessageRequest{
    string message = 1;
    uint64 user_id = 2;
    uint32 id = 3; 

};



service MessageService{

    rpc CreateMessage(CreateMessageRequest) returns (CreateMessageResponse);
    rpc GetMessages(GetMessagesRequest) returns (GetMessagesResponse);
    rpc DeleteMessage(DeleteMessageRequest) returns (google.protobuf.Empty);
    rpc UpdateMessage(UpdateMessageRequest) returns(google.protobuf.Empty);

    
}

